packages/core/interfaces/src/index.ts

```diff
@@ -158,6 +165,60 @@ export interface UseLocale
   loading: ComputedProperty<boolean>;
 }
 
+export interface ProductGetters<PRODUCT, PRODUCT_FILTER> {
+  getName: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<string>;
+  getSlug: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<string>;
+  getPrice: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<number | null>;
+  getGallery: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<UiMediaGalleryItem[]>;
+  getVariants: (products: PRODUCT[] | Readonly<PRODUCT[]>, filters?: PRODUCT_FILTER) =>
+    ComputedProperty<PRODUCT[]> | Readonly<PRODUCT[]>;
+  getAttributes: (product: PRODUCT[] | Readonly<PRODUCT[]>, filters?: Array<string>) =>
+    ComputedProperty<Array<AgnosticProductAttribute>>;
+  getDescription: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<any>;
+  getCategories: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<string[]>;
+  getId: (product: PRODUCT | Readonly<PRODUCT>) => ComputedProperty<number>;
+}
+
+export interface CartGetters<CART, PRODUCT> {
+  getProducts: (cart: CART) => ComputedProperty<PRODUCT[]>;
```

Comment:
Shopware and many other commerce platforms have a CartItem entity instead of Product directly.
Though the product is the most obvious type of cart item, there is something like a coupon, etc. 
We just need to keep in mind that fact. Maybe a little bit deeper level of abstraction is needed here? We are going to face this problem shortly ;)

Reply:
Sure, there is also a CART_ITEM type, which I can use here.